### PR TITLE
add kubearmor projects for gsoc 2023

### PR DESCRIPTION
**Pull request by kranurag7**
- Add project ideas for KubeArmor for GSoC 2023

### CODE REVIEW

File: summerofcode/2023.md [hunk 1]
```diff
@@ -319,6 +322,26 @@ This project lends itself to GSoC due to the modular nature of the tasks which a
 - Difficulty: Hard
 - Upstream Issue (URL): https://github.com/knative-sandbox/eventing-kafka-broker/issues/2917
 
+### Kubearmor
+
+#### GitHub Action for KubeArmor
+- Description: Build a GitHub action to allow the usage of kubearmor in the CI
```
Comment (aliok) [inline summerofcode/2023.md:328]: @kranurag7 Can you add more description here? I saw there's a good amount of information in the upstream ticket but still it is encouraged to have a description here that talks about these for attracting contributors:
- Problem, its context and motivation for the idea
- Solution in a few sentences

Feel free to have a look at some of the other ideas listed in this doc.

File: summerofcode/2023.md [hunk 2]
```diff
@@ -319,6 +322,26 @@ This project lends itself to GSoC due to the modular nature of the tasks which a
...
+- Expected outcome: Find security gaps and recommend policies leveraging GitHub action. The action should be able to  generate a summary using baseline benchmark and then show the application based changes in the graphical mode.
+- Mentor(s): Ankur Kothiwal(Ankurk99), Anurag Kumar(kranurag7), Barun Acharya(daemon1024)
+- Expected project size: 175 Hours
+- Recommended Skills: Kubernetes, GitHub Actions
+- Difficulty: Medium
+- Upstream Issue (URL): https://github.com/kubearmor/KubeArmor/issues/1128
+
+#### Store Kubearmor policies in OCI registry
+- Description: Store kubearmor policies & host policies in OCI registry.
```
Comment (aliok) [inline summerofcode/2023.md:337]: Same comment as the previous idea

File: summerofcode/2023.md [hunk 3]
```diff
@@ -319,6 +322,26 @@ This project lends itself to GSoC due to the modular nature of the tasks which a
 - Difficulty: Hard
 - Upstream Issue (URL): https://github.com/knative-sandbox/eventing-kafka-broker/issues/2917
 
+### Kubearmor
+
+#### GitHub Action for KubeArmor
+- Description: Build a GitHub action to allow the usage of kubearmor in the CI
+- Expected outcome: Find security gaps and recommend policies leveraging GitHub action. The action should be able to  generate a summary using baseline benchmark and then show the application based changes in the graphical mode.
+- Mentor(s): Ankur Kothiwal(Ankurk99), Anurag Kumar(kranurag7), Barun Acharya(daemon1024)
```
Comment (aliok) [inline summerofcode/2023.md:330]: @kranurag7 Can a project maintainer listed in [project-maintainers.csv](https://github.com/cncf/foundation/blob/main/project-maintainers.csv#L1027) post a LGTM on the PR?

Comment (aliok) [inline summerofcode/2023.md:330]: And, can you add email addresses of these mentors?

**Review by aliok — COMMENTED** [review]
Thanks for the ideas @kranurag7 .

I've added a few comments.

**Comment by aliok** [comment]
hi @kranurag7 . I saw you pushed some new commits. Feel free to ping me for a new review when you think you addressed the earlier comments.

File: summerofcode/2023.md [hunk 4]
```diff
@@ -319,6 +322,29 @@ This project lends itself to GSoC due to the modular nature of the tasks which a
...
+The action should be able to  generate a summary using baseline benchmark and then show the application based changes in the graphical mode.
+- Mentor(s): Ankur Kothiwal(Ankurk99,  ankur DOT kothiwal99 AT gmail DOT com), Anurag Kumar(kranurag7, contact DOT anurag7 AT gmail DOT com), Barun Acharya(daemon1024, barun1024 AT gmail DOT com)
+- Expected project size: 175 Hours
+- Recommended Skills: Kubernetes, GitHub Actions
+- Difficulty: Medium
+- Upstream Issue (URL): https://github.com/kubearmor/KubeArmor/issues/1128
+
+#### Store Kubearmor policies in OCI registry
+
+- Description: Store kubearmor policies & host policies in OCI registry. This project also aims to publish kubearmor policies on [Artifact Hub](https://artifacthub.io/).
```
Comment (aliok) [inline summerofcode/2023.md:340]: Can you also update the description of this idea?

**Comment by Ankurk99** [comment]
Hey @aliok, Thanks for taking the time to review the PR. We have made the required changes and one of the maintainers from [project-maintainers.csv](https://github.com/cncf/foundation/blob/main/project-maintainers.csv?rgh-link-date=2023-03-06T19%3A39%3A53Z#L1027) will be reviewing it soon.
Just wanted to point out that this list is not updated ref: https://github.com/kubearmor/KubeArmor/blob/main/MAINTAINERS.md

**Review by nyrahul — APPROVED** [review]
LGTM

File: summerofcode/2023.md [hunk 5]
```diff
@@ -53,6 +53,9 @@ If you are a project maintainer and consider mentoring during the GSoC 2023 cycl
     + [Dataplane migration for Apache Kafka communications: From Vert.x to Project Loom](#dataplane-migration-for-apache-kafka-communications--from-vertx-to-project-loom)
     + [Porting Knative Serving to Microshift](#porting-knative-serving-to-microshift)
     + [Self-Balancing Knative Kafka Broker partitions](#self-balancing-knative-kafka-broker-partitions)
+  * [KubeArmor](#kubearmor)
+    + [GitHub Actions for KubeArmor](#github-action-for-kubearmor)
```
Comment (aliok) [inline summerofcode/2023.md:57]:
```suggestion
    + [GitHub Actions for KubeArmor](#github-actions-for-kubearmor)
```

**Comment by aliok** [comment]
Thanks @kranurag7 !

/lgtm

DCO is acting up though.
It says this:

> Commit sha: [f64b0cc](https://github.com/cncf/mentoring/pull/867/commits/f64b0ccc90d1326aaf4aca981ac4db16a956e602), Author: Anurag, Committer: GitHub; Expected "Anurag [81210977+kranurag7@users.noreply.github.com](mailto:81210977+kranurag7@users.noreply.github.com)", but got "Anurag [81210977+kranurag7@users.noreply.github.com](mailto:81210977+kranurag7@users.noreply.github.com)".

Can you touch the last commit please? I guess dropping "Co-authored-by: Ali Ok <aliok@redhat.com>" from the commit message would solve the issue.

**Comment by aliok** [comment]
/lgtm

Thanks

**Review by aliok — APPROVED** [review]

**Comment by aliok** [comment]
I've just sent you and other mentors an invite for GSoC.

cc @Ankurk99, @kranurag7, @daemon1024

I saw you're already in CNCF GSoC member list @daemon1024. You just need to login to https://summerofcode.withgoogle.com/ and accept some documents so that I can add you to 2023.

**Pull request merged**